mlfinlab/backtest_statistics/statistics.py

```diff
@@ -0,0 +1,233 @@
+"""
+Implements the book Chapter 14 on Backtest Statistics
+"""
+
+import pandas as pd
+import scipy.stats as ss
+import numpy as np
+
+
+def timing_of_flattening_and_flips(target_positions: pd.Series) -> pd.DatetimeIndex:
+    """
+    Snippet 14.1, page 197, Derives the timestamps of flattening or flipping
+    trades from a pandas series of target positions. 
+    
+    :param target_positions: (pd.Series) target position series with timestamps as indices
+    :return: (pd.DatetimeIndex) timestamps of trades flattening, flipping and last bet
+    """
+    empty_positions = target_positions[(target_positions==0)].index # Empty positions index
+    previous_positions = target_positions.shift(1) # Timestamps pointing at previous positions
+    previous_positions = previous_positions[(previous_positions!=0)].index # Index of positions where previous one wasn't empty
+    flattening = empty_positions.intersection(previous_positions) # FLATTENING - if previous position was open, but current is empty
+    multiplied_posions = target_positions.iloc[1:] * target_positions.iloc[:-1].values # Multiplies current position with value of next one
+    flips = multiplied_posions[(multiplied_posions<0)].index # FLIPS - if current position has another direction compared to the next
+    res = flattening.union(flips).sort_values()
```

Comment:
Better variable names, please.

Reply:
I'll change _res_ to flips_and_flattening, as it was just a group of both. I was a bit puzzled as haven't found Lopez de Prado mentioning flattening in other parts of the book.